### PR TITLE
Add appapi commands to Android

### PR DESCRIPTION
**Pull request by corrm**
Add some commands to android payload.
payload pull request https://github.com/rapid7/metasploit-payloads/pull/336
,,
This pull for the metasploit UI/Extensions

## Verification
List the steps needed to make sure this thing works

- [x] Start `msfconsole`
- [x] `use exploit/multi/handler`
- [x] `set PAYLOAD android/meterpreter/...`
```
meterpreter > help
CorrM: Applictions Commands
===========================

    Command        Description
    -------        -----------
    app_install    Request to install apk file
    app_list       List installed apps in android device
    app_run        Start Main Activty for package name
    app_uninstall  Request to uninstall


Android Commands
================

    Command           Description
    -------           -----------
    activity_start    Start an Android activity from a Uri string
    check_root        Check if device is rooted
    dump_calllog      Get call log
    dump_contacts     Get contacts list
    dump_sms          Get sms messages
    geolocate         Get current lat-long using geolocation
    hide_app_icon     Hide the app icon from the launcher
    interval_collect  Manage interval collection capabilities
    send_sms          Sends SMS from target session
    set_audio_mode    Set Ringer Mode
    sqlite_query      Query a SQLite database from storage
    wakelock          Enable/Disable Wakelock
    wlan_geolocate    Get current lat-long using WLAN information
```

### CODE REVIEW

File: lib/rex/post/meterpreter/extensions/stdapi/corrm/apps.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# -*- coding: binary -*-
```
Comment (timwr) [inline lib/rex/post/meterpreter/extensions/stdapi/corrm/apps.rb]: Why don't we name this file `lib/rex/post/meterpreter/extensions/appapi/appapi.rb`?

**Comment by timwr** [comment]
Excellent contribution @CorrM . The code looks great.
I posted here already: https://github.com/rapid7/metasploit-payloads/pull/336
but I think this extension should be called `appapi` (matching stdapi, extapi) instead of `corrm`.
I can implement app_list and app_run on iOS.
Thoughts?

File: lib/rex/post/meterpreter/extensions/stdapi/tlv.rb [hunk 2]
```diff
@@ -5,33 +5,33 @@ module Post
 module Meterpreter
 module Extensions
 module Stdapi
-
+
```
Comment (timwr) [inline lib/rex/post/meterpreter/extensions/stdapi/tlv.rb]: Please try only include the necessary changes to this file (no whitespace changes).
Personally I use `git add -p` to cherry pick changes.

**Comment by CorrM** [comment]
i think i resolve every thing, and i see not it's so clear.
about IOS, i don't have and apple product so i don't interested to learn any thing about IOS. but if u can implement it. just do it 👍 

and what about Egypt in Metasploit. i see it a lot xD.

i will commit after some time

**Comment by CorrM** [comment]
Check this commit and i only need to know how to tell metasploit to know there is a `appapi` in `/rex/post/meterpreter/extensions`. (can u help)

**Comment by timwr** [comment]
Ah sorry for the confusion. We use this:
https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/base/sessions/meterpreter_android.rb#L26
to load the android commands automatically.
You can add this make it load the new extension:

```
--- a/lib/msf/base/sessions/meterpreter_android.rb
+++ b/lib/msf/base/sessions/meterpreter_android.rb
@@ -24,6 +24,7 @@ class Meterpreter_Java_Android < Msf::Sessions::Meterpreter_Java_Java
     original = console.disable_output
     console.disable_output = true
     console.run_single('load android')
+    console.run_single('load appapi')
     console.disable_output = original
   end
```

**Comment by timwr** [comment]
Apologies my first comment was clear as mud.
Please rename the commands from `android_app_run` to `appapi_app_run` etc.
Copy the android extension (or python) in
lib/rex/post/meterpreter/ui/console/command_dispatcher
and
lib/rex/post/meterpreter/extensions/
but name the extension `appapi`.
(Don't copy the stdapi or extapi extension, as they are unnecessarily complicated).

If you get stuck I can send you a pull request :)

**Comment by timwr** [comment]
@CorrM this is a good example of adding a simple extension to the framework: https://github.com/rapid7/metasploit-framework/commit/bba7bcc8851d17402a2db49d487b9bf54558c5cb#diff-0baa630913d7a97a47ef02c7865a7baf

**Comment by CorrM** [comment]
@timwr ty for all this info. can u wait me one hour or less i will push my commit. <3

**Comment by timwr** [comment]
No problem thanks for your contribution! <3
Don't rush, I'll hopefully get around to testing it tomorrow.

**Comment by CorrM** [comment]
@timwr any thing about ext color .? like
```
AdBlocker Reborn                    com.aviraxp.adblocker.continued            false    false =====> Red text
AllowLongSMS                        uk.co.villainrom.pulser.allowlongsms       false    false =====> Blue text
```

**Comment by timwr** [comment]
I would just leave it black and white. I guess you could add color as an option though, hardly seems worth it.

**Comment by CorrM** [comment]
i think we are ready to go 👍

**Comment by timwr** [comment]
Ping @eqypt

**Comment by egypt** [comment]
Pong @timwr

**Comment by CorrM** [comment]
so whats news

**Comment by CorrM** [comment]
I think the last commit changes better than before.  👍 (Payload updated too)

**Review by wvu — CHANGES_REQUESTED** [review]
There are still traces of the string `corrm`, and there are still unrelated whitespace changes. Please fix those. This is looking a lot better already!

**Comment by CorrM** [comment]
> There are still traces of the string `corrm`, and there are still unrelated whitespace changes. Please fix those. This is looking a lot better already!

i did'n see `corrm` in my file changed in github.
so i my pull will accept i think it's the time for that.

**Review by corrm — COMMENTED** [review]
ok, it's good to remove white spaces

File: lib/rex/post/meterpreter/extensions/stdapi/tlv.rb (hunk 2 above)
Comment (wvu-r7) [reply lib/rex/post/meterpreter/extensions/stdapi/tlv.rb]: These are still here.

**Comment by CorrM** [comment]
@timwr hey dude what's knows i think there is no matter to get some new stuff in metasploit !!.

**Comment by timwr** [comment]
@CorrM please see https://github.com/CorrM/metasploit-framework/pull/1/files and https://github.com/CorrM/metasploit-payloads/pull/1/files

If you merge them, the changes should appear in this branch. Let me know if I broke something. I removed the root install/uninstall method, let's add it later.

**Comment by CorrM** [comment]
@timwr np i will merge, but why removed root stuff .?

**Comment by timwr** [comment]
I'm not sure the root stuff belongs in the payloads repository as all its doing is calling `pm install` with su. I think we should be doing that from the framework repo.

**Comment by timwr** [comment]
I'm not sure the root stuff belongs in the payloads repository as all its doing is calling `pm install` with su. I think we should be doing that from the framework repo.

**Comment by CorrM** [comment]
Accepted.

**Review by wvu — APPROVED** [review]
Good enough, thanks.

**Comment by timwr** [comment]
Excellent work @CorrM 
Many thanks!

**Comment by CorrM** [comment]
welcome <3, wait me next work :D

**Comment by timwr** [comment]
Ping me on slack or IRC if you have questions. I'm looking forward to your next change. <3

**Comment by tdoan-r7** [comment]
# Release Notes
The Android payload now has new addapi commands.

**Pull request merged**